index.js

```diff
@@ -308,15 +308,50 @@ SendStream.prototype.hasTrailingSlash = function hasTrailingSlash () {
 }
 
 /**
+ * Returns `true` if any request preconditions are failing
+ *
+ * @return {Boolean}
+ * @api private
+ */
+
+SendStream.prototype.isFailingPrecondition = function isFailingPrecondition (stat) {
+  // if-match
+  var match = this.req.headers['if-match']
+  var etag = this.res._headers['etag']
+  var strongEtag = /^(?!W\/)/
+
+  if (match) {
+    if (strongEtag.test(etag)) {
```

Comment:
From my reading of RFC 7232, `If-Match: "foo"` should fail with 412 if this module is about to respond with `ETag: W/"bar"`, but this will not fail here since it skips matching if this module is about to send a weak ETag, which seems to not match the requirements.